src/components/AChat/AChat.vue

```diff
@@ -42,28 +42,31 @@
   </div>
 </template>
 
-<script>
-import { ref, onMounted, onBeforeUnmount, computed } from 'vue'
+<script lang="ts">
+import { ref, onMounted, onBeforeUnmount, computed, Ref, defineComponent } from 'vue'
 import throttle from 'lodash/throttle'
 import scrollIntoView from 'scroll-into-view-if-needed'
 import Styler from 'stylefire'
 import { animate } from 'popmotion'
 import { SCROLL_TO_REPLIED_MESSAGE_ANIMATION_DURATION } from '@/lib/constants'
 import { isStringEqualCI } from '@/lib/textHelpers'
 import { isWelcomeChat } from '@/lib/chat/meta/utils/isWelcomeChat'
+import { NormalizedChatMessageTransaction } from '@/lib/chat/helpers'
+import { User } from '@/components/AChat/types.ts'
 
-export default {
+export default defineComponent({
   props: {
     messages: {
-      type: Array,
+      type: Array as () => NormalizedChatMessageTransaction[],
```

Comment:
I think you should use generic `PropType` here. Example:

https://github.com/Adamant-im/adamant-im/blob/4bc956c1ff2e96b4f08091b2fc07c2bf062d2678/src/components/AChat/AChatAttachment/AChatAttachment.vue#L139

Check other places as well.